docs/tutorial/t-spark-streaming.md

```diff
@@ -0,0 +1,335 @@
+  --properties-file properties.conf
+```
+
+Now, let's create a minimal Kafka and Zookeeper setup. This can be done both easily and quickly using [`zookeeper-k8s`](https://github.com/canonical/zookeeper-k8s-operator) and [`kafka-k8s`](https://charmhub.io/kafka-k8s) charms. Single units of these charms running in our model should be enough.
+
+```bash
+# Deploy Zookeper
+juju deploy zookeeper-k8s --series=jammy --channel=edge
+
+# Deploy Kafka
+juju deploy kafka-k8s --series=jammy --channel=edge
+```
+
+Once installed, let's see the current status of the Juju model with the following command:
+```bash
+juju status --watch 1s
+```
+
+This command periodically refreshes status of Juju model and then present the data in the console. Once all the charms have been deployed (you may need to wait some time), the output should appear similar to the following:
+
+```
+Model            Controller      Cloud/Region        Version  SLA          Timestamp
+spark-streaming  spark-tutorial  microk8s/localhost  3.1.7    unsupported  10:09:40Z
+
+App            Version  Status   Scale  Charm          Channel  Rev  Address         Exposed  Message
+kafka-k8s               waiting      1  kafka-k8s      3/edge    47  10.152.183.242  no       installing agent
+zookeeper-k8s           active       1  zookeeper-k8s  3/edge    42  10.152.183.87   no       
+
+Unit              Workload  Agent  Address      Ports  Message
+kafka-k8s/0*      blocked   idle   10.1.29.184         missing required zookeeper relation
+zookeeper-k8s/0*  active    idle   10.1.29.182         
+```
+
+The `kafka-k8s/0` unit is blocked because we have not integrated Kafka with Zookeeper yet. We can do that using:
+
+```bash
+juju integrate kafka-k8s zookeeper-k8s
+```
+
+After the integration, `juju status` should have an output similar to the following:
+
+```
+Model            Controller      Cloud/Region        Version  SLA          Timestamp
+spark-streaming  spark-tutorial  microk8s/localhost  3.1.7    unsupported  10:13:55Z
+
+App            Version  Status  Scale  Charm          Channel  Rev  Address         Exposed  Message
+kafka-k8s               active      1  kafka-k8s      3/edge    47  10.152.183.242  no       
+zookeeper-k8s           active      1  zookeeper-k8s  3/edge    42  10.152.183.87   no       
+
+Unit              Workload  Agent  Address      Ports  Message
+kafka-k8s/0*      active    idle   10.1.29.184         
+zookeeper-k8s/0*  active    idle   10.1.29.182 
+```
+
+As you can see, both Kafka and Zookeeper charms are in "active" status. However, it can take some amount of time before the app and units are finally transitioned to active state.
+
+For us to experiment with the streaming feature, we would want some sample streaming data to be generated in Kafka continuously in real time. For that, we can use the `kafka-test-app` charm as a producer of events. Let's deploy the charm, and integrate it with `kafka-k8s` so that it is able to write messages to Kafka.
+
```

Comment:
```suggestion
For us to experiment with the streaming feature, we would want some sample streaming data to be generated in Kafka continuously in real time. For that, we can use the `kafka-test-app` charm as a producer of events. 

Let's deploy the charm, and integrate it with `kafka-k8s` so that it is able to write messages to Kafka.
```